sematic/config/user_settings.py

```diff
@@ -0,0 +1,306 @@
+# Standard Library
+import distutils.util
+import enum
+import logging
+import os
+from dataclasses import asdict, dataclass
+from typing import Any, Dict, Optional
+
+# Third-party
+import yaml
+
+# Sematic
+from sematic.config.config_dir import get_config_dir
+
+logger = logging.getLogger(__name__)
+
+_USER_SETTINGS_FILE = "settings.yaml"
+_DEFAULT_PROFILE = "default"
+
+
+class UserSettingsVar(enum.Enum):
+    # Server:
+    # Sematic
+    SEMATIC_AUTHENTICATE = "SEMATIC_AUTHENTICATE"
+    SEMATIC_AUTHORIZED_EMAIL_DOMAIN = "SEMATIC_AUTHORIZED_EMAIL_DOMAIN"
+    SEMATIC_WORKER_API_ADDRESS = "SEMATIC_WORKER_API_ADDRESS"
+
+    # Google
+    GOOGLE_OAUTH_CLIENT_ID = "GOOGLE_OAUTH_CLIENT_ID"
+
+    # Github
+    GITHUB_OAUTH_CLIENT_ID = "GITHUB_OAUTH_CLIENT_ID"
+
+    # Kubernetes
+    KUBERNETES_NAMESPACE = "KUBERNETES_NAMESPACE"
+
+    # GRAFANA
+    GRAFANA_PANEL_URL = "GRAFANA_PANEL_URL"
+
```

Comment:
Isn't it strange to have server settings here? I thought we would separate them? Different enum maybe?
This module is named `user_settings` and contains server settings...

Reply:
This PR prepares the codebase for splitting out server settings, as stated in the description. All these mixed settings existed as-is in the `SettingsVar` enum. Here I rename it to `UserSettingsVar` and group them into 2 sections, just for my convenience. The next PR will remove the server section.